startup/gui/cyclesViewerSettings.gfr

```diff
@@ -0,0 +1,316 @@
+__children["ViewerSettings"].addChild( Gaffer.NameSwitch( "ShadingSwitch" ) )
+__children["ViewerSettings"]["ShadingSwitch"].setup( GafferScene.ScenePlug( "value", ) )
+__children["ViewerSettings"]["ShadingSwitch"]["in"].addChild( Gaffer.NameValuePlug( "", GafferScene.ScenePlug( "value", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ), True, "in2", Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic ) )
+__children["ViewerSettings"]["ShadingSwitch"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ViewerSettings"].addChild( GafferScene.ShaderAssignment( "FlatAssignment" ) )
+__children["ViewerSettings"]["FlatAssignment"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ViewerSettings"].addChild( Gaffer.Dot( "Dot" ) )
+__children["ViewerSettings"]["Dot"].setup( GafferScene.ScenePlug( "in", ) )
+__children["ViewerSettings"]["Dot"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ViewerSettings"].addChild( Gaffer.Dot( "Dot1" ) )
+__children["ViewerSettings"]["Dot1"].setup( GafferScene.ScenePlug( "in", ) )
+__children["ViewerSettings"]["Dot1"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ViewerSettings"].addChild( GafferScene.ShaderAssignment( "PreviewAssignment" ) )
+__children["ViewerSettings"]["PreviewAssignment"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ViewerSettings"].addChild( Gaffer.StringPlug( "lighting", defaultValue = 'direct', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ViewerSettings"].addChild( Gaffer.Backdrop( "Backdrop" ) )
+__children["ViewerSettings"]["Backdrop"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ViewerSettings"]["Backdrop"].addChild( Gaffer.Box2fPlug( "__uiBound", defaultValue = imath.Box2f( imath.V2f( -10, -10 ), imath.V2f( 10, 10 ) ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["ViewerSettings"].addChild( Gaffer.Spreadsheet( "LightingSpreadsheet" ) )
+__children["ViewerSettings"]["LightingSpreadsheet"]["rows"].addColumn( Gaffer.NameValuePlug( "ai:GI_total_depth", Gaffer.IntPlug( "value", defaultValue = 10, ), False, "giTotalDepth", Gaffer.Plug.Flags.Default ), adoptEnabledPlug = True )
+__children["ViewerSettings"]["LightingSpreadsheet"]["rows"].addColumn( Gaffer.IntPlug( "maxBounceValue", defaultValue = 7, ) )
```

Comment:
This would be better as a NameValuePlug as currently it overrides to 0 when `lighting` is set to "full".

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/4807/commits/2f9a527197c55a859085af2f4a9c6cb5fc1000cb